src/server/scripts/Events/hallows_end.cpp

```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+                    case EVENT_FINISH:
+                    {
+                        bool failed = false;
+                        std::list<Creature*> triggerList;
+                        me->GetCreatureListWithEntryInGrid(triggerList, NPC_FIRE_TRIGGER);
+                        for (Creature* trigger : triggerList)
+                        {
+                            if (trigger->HasAura(SPELL_FIRE_AURA_BASE))
+                                failed = true;
+
+                            trigger->AI()->EnterEvadeMode();
+                        }
+                        me->StopMoving();
+                        FinishEvent(failed);
+                        break;
+                    }
+                    case EVENT_CLEAVE:
+                        me->GetAI()->DoCast(SPELL_CLEAVE);
+                        events.Repeat(Seconds(20));
+                        break;
+                }
+
+            }
+
+            if (!UpdateVictim())
+                return;
+
+            DoMeleeAttackIfReady();
+        }
+
+        void FinishEvent(bool failed)
+        {
+            events.Reset();
+            if (failed)
+            {
+                Talk(3);
+                me->DespawnOrUnsummon();
+            }
+            else
+            {
+                Talk(4);
+                float x, y, z;
+                GetPosToLand(x, y, z);
+                me->GetMotionMaster()->MovePoint(1, x, y, z);
+            }
+        }
+
+        void MovementInform(uint32 type, uint32 point) override
+        {
+            if (type == POINT_MOTION_TYPE && point == 1)
+            {
+                me->SetUInt32Value(UNIT_FIELD_FLAGS, 0);
+                me->SetDisableGravity(false);
+                me->RemoveAura(SPELL_HORSEMAN_MOUNT);
+                //me->Dismount();
+                if (Unit* target = me->SelectNearestPlayer(30.0f))
+                    AttackStart(target);
+            }
+        }
+
+        void JustDied(Unit* killer) override
+        {
+            Talk(5);
+            float x, y, z;
+            GetPosToLand(x, y, z);
+            me->CastSpell(x, y, z, SPELL_SUMMON_LANTERN, true);
+            CompleteQuest();
+        }
+
+        void CompleteQuest()
+        {
+            float radius = 100.0f;
+            std::list<Player*> players;
+            Trinity::AnyPlayerInObjectRangeCheck checker(me, radius);
+            Trinity::PlayerListSearcher<Trinity::AnyPlayerInObjectRangeCheck> searcher(me, players, checker);
+            me->VisitNearbyWorldObject(radius, searcher);
+
+            for (std::list<Player*>::const_iterator itr = players.begin(); itr != players.end(); ++itr)
+            {
+                (*itr)->AreaExploredOrEventHappens(QUEST_STOP_THE_FIRES_H);
+                (*itr)->AreaExploredOrEventHappens(QUEST_STOP_THE_FIRES_A);
+                (*itr)->AreaExploredOrEventHappens(QUEST_LET_THE_FIRES_COME_H);
+                (*itr)->AreaExploredOrEventHappens(QUEST_LET_THE_FIRES_COME_A);
```

Comment:
Nope, there's no required creature kill in quest_template for those quests, sadly.